packages/editor-ui/src/components/mixins/workflowHelpers.ts

```diff
@@ -330,24 +330,31 @@ export const workflowHelpers = mixins(
 					if (node.credentials !== undefined && nodeType.credentials !== undefined) {
 						const saveCredenetials: INodeCredentials = {};
 						for (const nodeCredentialTypeName of Object.keys(node.credentials)) {
-							// todo revert to only set actually used credentials on workflow
-							// if (this.hasProxyAuth(node) || Object.keys(node.parameters).includes('genericAuthType')) {
-							// 	saveCredenetials[nodeCredentialTypeName] = node.credentials[nodeCredentialTypeName];
-							// 	continue;
-							// }
-
-							// const credentialTypeDescription = nodeType.credentials
-							// 	.find((credentialTypeDescription) => credentialTypeDescription.name === nodeCredentialTypeName);
-
-							// if (credentialTypeDescription === undefined) {
-							// 	// Credential type is not know so do not save
-							// 	continue;
-							// }
-
-							// if (this.displayParameter(node.parameters, credentialTypeDescription, '', node) === false) {
-							// 	// Credential should not be displayed so do also not save
-							// 	continue;
-							// }
+							if (this.hasProxyAuth(node) || Object.keys(node.parameters).includes('genericAuthType')) {
+								saveCredenetials[nodeCredentialTypeName] = node.credentials[nodeCredentialTypeName];
+								continue;
+							}
+
+							const credentialsByVersion = nodeType.credentials.filter(c => {
+								if (c.displayOptions && c.displayOptions.show && c.displayOptions.show['@version']) {
```

Comment:
should not this be handled by `this.displayParameter`.. I think we need a deeper fix here... this is a bandaid solution